nms/app/views/domain-proxy/__tests__/CbsdEdit.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2022 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import MuiStylesThemeProvider from '@material-ui/styles/ThemeProvider';
+import React from 'react';
+import defaultTheme from '../../../theme/default';
+import {AddEditCbsdButton, CbsdAddEditDialog} from '../CbsdEdit';
+
+import MagmaAPIBindings from '../../../../generated/MagmaAPIBindings';
+import {CbsdContextProvider} from '../../../components/lte/LteContext';
+import {MuiThemeProvider} from '@material-ui/core/styles';
+import {fireEvent, render, waitFor, within} from '@testing-library/react';
+
+jest.mock('../../../../generated/MagmaAPIBindings.js');
+
+const mockEnqueueSnackbar = jest.fn();
+jest.mock('../../../hooks/useSnackbar', () => ({
+  useEnqueueSnackbar: () => mockEnqueueSnackbar,
+}));
+
+const cbsd = {
+  capabilities: {
+    antenna_gain: 0,
+    max_power: 24,
+    min_power: 0,
+    number_of_antennas: 1,
+  },
+  cbsd_category: 'b',
+  cbsd_id: '2AG32PBS31010/1202000291213VB0009',
+  desired_state: 'registered',
+  fcc_id: '2AG32PBS31010',
+  frequency_preferences: {
+    bandwidth_mhz: 20,
+    frequencies_mhz: [3600],
+  },
+  id: 28,
+  is_active: false,
+  serial_number: '1202000291213VB0009',
+  single_step_enabled: false,
+  state: 'unregistered',
+  user_id: 'SAS-Freedomfi',
+};
+
+const networkId = 'test-network';
+
+/**
+ * Convert cbsd that we get from /cbsds endpoint (that is mocked in the test) to cbsd payload for updating or creating cbsd
+ */
+const convertCbsdToPayload = cbsd => {
+  const payload = {...cbsd};
+  delete payload.is_active;
+  delete payload.id;
+  delete payload.state;
+  delete payload.cbsd_id;
+  return payload;
+};
+
+const renderWithProviders = jsx => {
+  return render(
+    <MuiThemeProvider theme={defaultTheme}>
+      <MuiStylesThemeProvider theme={defaultTheme}>
+        <CbsdContextProvider networkId={networkId}>{jsx}</CbsdContextProvider>
+      </MuiStylesThemeProvider>
+    </MuiThemeProvider>,
+  );
+};
+
+describe('<AddEditCbsdButton />', () => {
+  beforeEach(() => {
+    MagmaAPIBindings.getDpByNetworkIdCbsds.mockResolvedValue({
+      cbsds: [cbsd],
+      total_count: 1,
+    });
+  });
+
+  afterEach(() => {
+    MagmaAPIBindings.getDpByNetworkIdCbsds.mockClear();
```

Comment:
We no longer need to do this manually, this now happens automatically.

Reply:
Hm indeed it works without this line.
I find it confusing though, if I mock resolved value in `beforeEach`, I always reset it in `afterEach`.
So the code becomes more weird without explicit resets.

Do you really want me to remove this `mockClear()` ? :)